metricbeat/module/vsphere/resourcepool/data.go

```diff
@@ -0,0 +1,55 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package resourcepool
+
+import (
+	"github.com/vmware/govmomi/vim25/mo"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func (m *ResourcePoolMetricSet) mapEvent(rp mo.ResourcePool, data *metricData) mapstr.M {
+	event := mapstr.M{
+		"name":   rp.Name,
+		"status": rp.OverallStatus,
+	}
+
+	if rp.Summary.GetResourcePoolSummary().QuickStats != nil {
```

Comment:
You could use the result of calling `GetResourcePoolSummary().QuickStats` and return early.  
```suggestion
	quickStats := rp.Summary.GetResourcePoolSummary().QuickStats 
	if quickStats == nil {
	    return event
	}    
```
And also use them in the all `event.Put`s